.github/workflows/nightly-emscripten.yml

```diff
@@ -0,0 +1,120 @@
+name: Emscripten nightly build + solc-bin push
+
+on:
+  schedule:
+    # Run once a day, at midnight
+    - cron: '0 0 * * *'
+
+jobs:
+  build-emscripten:
+    runs-on: ubuntu-latest
+    outputs:
+      nightly-version: ${{ env.NIGHTLY_VERSION }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/solidity'
+          ref: 'develop'
+
+      - name: Build soljson.js
+        run: |
+          # Note that this script will spawn and build inside a docker image (which works just fine in github actions).
+          scripts/build_emscripten.sh
+
+      - name: Get nightly version
+        id: nightly-version
+        run: |
+          last_commit_timestamp=$(git log -1 --date=iso --format=%ad HEAD)
+          last_commit_date=$(date --date="$last_commit_timestamp" --utc +%Y.%-m.%-d)
+          last_commit_hash=$(git rev-parse --short=8 HEAD)
+          solidity_version=$(scripts/get_version.sh)
+          nightly_version="v${solidity_version}-nightly.${last_commit_date}+commit.${last_commit_hash}"
+
+          echo "::set-env name=NIGHTLY_VERSION::${nightly_version}"
+
+      - name: Upload soljson.js as an artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: soljson.js
+          path: upload/soljson.js
+
+  commit-soljson:
+    runs-on: ubuntu-latest
+    needs: build-emscripten
+
+    env:
+      COMMITTER_NAME: emscripten nightly action
+      # TODO: Set a proper e-mail address
+      COMMITTER_EMAIL: solidity@example.com
+      TARGET_BRANCH: master
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      NIGHTLY_VERSION: ${{ needs.build-emscripten.outputs.nightly-version }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ env.TARGET_BRANCH }}
+
+      - name: Make sure that the working copy is clean
+        run: |
+          echo "::debug::$(git status)"
+          test -z "$(git status --porcelain)" || exit 1
```

Comment:
Do we really need this? We just checked it out three lines up, didn't we? Or was this only for you testing stuff, then nevermind :-).

Reply:
I'm just being paranoid because to the file truncations :) This check is here to prove that files we're getting out of `git clone` are still clean. I treat it more like an assertion. Ideally, I'd put it at the end of the previous step but I don't think I can do that.

`scripts/travis-emscripten/publish_binary.sh` actually does something very similar. It runs `git clean` immediately after a fresh clone and checkout to make sure the tree is clean. It shouldn't be necessary unless Travis keeps files between runs (does it?):
```
git clone --depth 2 git@github.com:ethereum/solc-bin.git
cd solc-bin
git config user.name "travis"
git config user.email <email>
git checkout -B gh-pages origin/gh-pages
git clean -f -d -x
```

Reply:
Fair enough - a little paranoia doesn't hurt I guess!